Source/Engine/AGS.Engine/Audio/ALAudioClip.cs

```diff
@@ -23,13 +24,16 @@ public ALAudioClip(string id, ISoundData soundData, IAudioSystem system, IAudioE
 			_errors = errors;
             _backend = backend;
 			ID = id;
-			_buffer = new Lazy<int> (() => generateBuffer());
+            _duration = AudioHelpers.GetDuration(soundData.DataLength, soundData.Channels, soundData.BitsPerSample, soundData.SampleRate);
+            _buffer = new Lazy<int> (() => generateBuffer());
             _playingSounds = new AGSConcurrentHashSet<ISound>();
 			_system = system;
 			Volume = 1f;
 			Pitch = 1f;
 		}
 
+        public float Duration { get => _duration; }
```

Comment:
Instead of public property + member, you can just have `public float Duration { get; private set; }` so no need for the member.